src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -203,6 +203,13 @@ describe('DOMPropertyOperations', () => {
       expect(stubNode.getAttribute('role')).toBe('<html>');
     });
 
+    it('should set values as boolean attributes', () => {
+      DOMPropertyOperations.setValueForProperty(stubNode, 'data-foo', true);
+      expect(stubNode.getAttribute('data-foo')).toBe('');
+      DOMPropertyOperations.setValueForProperty(stubNode, 'data-foo', false);
+      expect(stubNode.getAttribute('data-foo')).toBe(null);
+    });
+
```

Comment:
That is correct; `DOMStringMap` (the backing type for `.dataset`) always only holds string values.